coralproject/talk
# reset password endpoints

## What is this?

We're using JWTs to verify the user and enable resetting user passwords from the API. This PR adds the functionality to do this.

## How to test

Since the UI is not in master for this, you'll just have to use `curl` or Postman.

- POST to `http://localhost:3000/api/v1/user/request-password-reset` with the body of the request as json with `{"email": "your_email"}`
- if your email is associated with a local profile in the db, you will get an email from coralcore@mozillafoundation.org at that address with a link to reset your password. If no user was found, a 500 is thrown
- the big token in the email only shows up if `NODE_ENV != 'production'`
- copy that token
- make a POST request to `http://localhost:3000/api/v1/user/update-password` with a json payload of `{"token": "the_token", "password": "new_password"}`
- you should get a 204 response that your password was updated.

I have no idea how to test the email component of this PR.
